Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -397,6 +397,19 @@ def visit_MatchNode(self, node):
         self.visitchildren(node)
         return node
 
+    def visit_MatchValuePatternNode(self, node):
+        if isinstance(node.value, ExprNodes.JoinedStrNode):
+            error(node.value.pos, "f-strings are not accepted for pattern matching")
+        self.visitchildren(node)
+        return node
```

Comment:
That is currently rejected. The structure is

```
- MatchSequencePatternNode
   - patterns: [
       - MatchValuePatternNode
          - value: JoinedStrNode
    ]
```

so it does just rejected by the check in `visit_MatchValuePatternNode`. (`MatchMappingPatternNode` is special-cased because the keys aren't made up of `MatchValuePatternNodes`. That's mainly because the rules for matching keys are a little difference (they're all match by `==` rather than some being matched by `is`) so there wasn't much shared implementation).

I'll switch to a status flag on my next lot of edits though - that's probably future-proof it a little more.

Reply:
This is not done by a status flag in `PostParse`. I don't think it changes the behaviour about what's allowed, but it's probably a little clearer (no `isinstance` check except the hidden one in the visitor dispatch)